redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java

```diff
@@ -105,25 +117,42 @@ public DatabaseConfig delete(String name, String environmentId) {
     }
 
     private DatabaseConfig deleteOne(DatabaseConfig databaseConfig) {
+        LOGGER.debug("Deleting database with name: {}", databaseConfig.getName());
         try {
+            // FIXME this can take a while and trigger logging that the transaction took too long
             return transactionService.required(() -> {
-                delete(databaseConfig);
-                if (databaseConfig.isUserManaged()) {
-                    deletePhysicalDatabase(databaseConfig);
+                if (databaseConfig.getStatus() == ResourceStatus.SERVICE_MANAGED) {
+                    return deleteServiceManagedDatabase(databaseConfig);
+                } else {
+                    return delete(databaseConfig);
```

Comment:
I think in any case we should delete the entry from databaseConfig table, so it should not go to the else branch.

Reply:
Right, that happens. deleteServiceManagedDatabase calls delete once the database has been removed from the server.